Add containers and configure environment

ingluife opened the pull request:
Hi, @flyve-mdm/backend-development .

This is the configuration of all docker containers and the environment.
I am checking the issue in the git clone and the composer install for GLPI application.

Closes #21

ingluife committed:
build(docker): configure all containers

Signed-off-by: Luis F. Gonzalez <ing.luisgonzalez23@gmail.com>

ingluife committed:
build(docker): add folder empty

Signed-off-by: Luis F. Gonzalez <ing.luisgonzalez23@gmail.com>

ingluife committed:
build(docker): add file environment variables

Signed-off-by: Luis F. Gonzalez <ing.luisgonzalez23@gmail.com>

ingluife committed:
build(docker): configure clone repo GLPI

Signed-off-by: Luis F. Gonzalez <ing.luisgonzalez23@gmail.com>

ingluife committed:
build(docker): delete layers in container

edit dockerfile php

Signed-off-by: Luis F. Gonzalez <ing.luisgonzalez23@gmail.com>

ingluife committed:
build(docker): modific nodejs version

disabled git sslVerific

Signed-off-by: Luis F. Gonzalez <lgonzalez@teclib.com>

ingluife committed:
build(docker): edit variables to clone

Signed-off-by: Luis F. Gonzalez <lgonzalez@teclib.com>

ingluife committed:
build(docker): change the form to clone

add validation if www is empty

Signed-off-by: Luis F. Gonzalez <lgonzalez@teclib.com>

ingluife committed:
build(docker): ignore www directory

Signed-off-by: Luis F. Gonzalez <lgonzalez@teclib.com>

ingluife committed:
build(docker): add variable and command

change name of containers

Signed-off-by: Luis F. Gonzalez <lgonzalez@teclib.com>

ingluife committed:
build(docker): change cliinstall

add command in compose
change directory of www-data
add permission in directories after cloning
composer install as www-data user

Signed-off-by: Luis F. Gonzalez <lgonzalez@teclib.com>

ingluife committed:
build(docker): change web server

add apache directory
configure vhost
configure Dockerfile

Signed-off-by: Luis F. Gonzalez <lgonzalez@teclib.com>

ajsb85 commented:
Hi, @ingluife 
There are commits without signature and others with signature unverified.

ajsb85 reviewed (changes requested):
Hi, @ingluife 
There are commits without signature and others with signature unverified.

ingluife committed:
build(docker): add config apache and site

enable proxy fcgi for php-fpm

Signed-off-by: Luis F. Gonzalez <lgonzalez@teclib.com>

ingluife committed:
build(docker): add configuration http.conf

Add VirtualHost
Configure cron for GLPI
install cron Dependency
Change root apache

Signed-off-by: Luis F. Gonzalez <lgonzalez@teclib.com>

ajsb85 commented on .env:
```diff
@@ -0,0 +1,16 @@
+#Enviroment GLPI
+CLONNING_METHOD=ssh
+GLPI_BRANCH=9.2/bugfixes
+GLPI_SOURCE=git@github.com:glpi-project/glpi.git
```
Hi, @ingluife 
Maybe a new env var, for the server name: `dev.glpi.local`

ingluife replied:
Good, i will add it as environment variable.

ajsb85 commented on docker/php/custom.ini:
```diff
@@ -0,0 +1,29 @@
+; General settings
+date.timezone = "America/Caracas"
```
Hi, @ingluife 
Maybe this time zone can be the default and set up optionally with an external arguments.

ingluife replied:
Ok, perfect. Today i will do it. But default Time Zone should be UTC. Do you agree?

ingluife committed:
build(docker): add default timezone variable

add variable ports
configure directory name site variable
configure server_name variable
edit DocumentRoot in VirtualHost

Signed-off-by: Luis F. Gonzalez <lgonzalez@teclib.com>

ingluife commented:
Hi @flyve-mdm/backend-development.

This docker environment is ready for tested.
